lightning/src/chain/mod.rs

```diff
@@ -132,9 +132,34 @@ pub trait Filter: Send + Sync {
 	/// a spending condition.
 	fn register_tx(&self, txid: &Txid, script_pubkey: &Script);
 
-	/// Registers interest in spends of a transaction output identified by `outpoint` having
-	/// `script_pubkey` as the spending condition.
-	fn register_output(&self, outpoint: &OutPoint, script_pubkey: &Script);
+	/// Registers interest in spends of a transaction output.
+	///
+	/// Optionally, when `output.block_hash` is set, should return any transaction spending the
+	/// output that is found in the corresponding block at the given index.
+	///
+	/// This return value is useful for Electrum clients in order to supply in-block descendant
```

Comment:
I don't understand how the Electrum client is leveraging block hash presence. Do you assume it stores the in-block descendant, even before `register_output` caller ask for them ?

Otherwise, I don't think you need block hash to register already-confirmed scriptpubkeys to an Electrum server API, see https://electrumx.readthedocs.io/en/latest/protocol-methods.html